src/main/java/core/basesyntax/dao/StoreCsvDaoImpl.java

```diff
@@ -0,0 +1,149 @@
+        }
+        try (InputStream inputStream = new FileInputStream(file);
+                InputStreamReader inputStreamReader = new InputStreamReader(
+                        inputStream, StandardCharsets.UTF_8);
+                BufferedReader reader = new BufferedReader(inputStreamReader)) {
+            reader.readLine();
+            String line;
+            while ((line = reader.readLine()) != null) {
+                lines.add(line);
+            }
+        } catch (IOException e) {
+            throw new RuntimeException("Cannot read file"
+                    + PATH_TO_DAILY_ACTIVITY_FILE, e);
+        }
+        return lines;
+    }
+
+    private void addColumnNames() {
+        if (isEmptyFile()) {
+            StringBuilder columnNamesBuilder = new StringBuilder();
+            columnNamesBuilder.append("type,");
+            columnNamesBuilder.append("fruit,");
+            columnNamesBuilder.append("quantity");
+            columnNamesBuilder.append(System.lineSeparator());
+            String columnNames = columnNamesBuilder.toString();
+            writeLinesToFile(columnNames, StoreCsvDaoImpl.PATH_TO_DAILY_ACTIVITY_FILE);
+            System.out.println("Columns was added successfully to file: "
+                    + PATH_TO_DAILY_ACTIVITY_FILE);
+        }
+    }
+
+    private boolean isEmptyFile() {
+        boolean isEmptyFile;
+        try (InputStream inputStream = new FileInputStream(PATH_TO_DAILY_ACTIVITY_FILE);
+                InputStreamReader inputStreamReader = new InputStreamReader(inputStream,
+                        StandardCharsets.UTF_8);
+                BufferedReader reader = new BufferedReader(inputStreamReader)) {
+            isEmptyFile = reader.readLine() == null;
+        } catch (IOException e) {
+            throw new RuntimeException("Cannot read file" + PATH_TO_DAILY_ACTIVITY_FILE, e);
+        }
+        return isEmptyFile;
+    }
+
+    private FruitTransaction getTransactionFromCsv(String line) {
+        Optional.ofNullable(line)
+                .orElseThrow(() -> new IllegalArgumentException(STRING_ARGUMENT_ERROR));
+        String[] row = line.split(",");
+        Operation operation = Operation.chooseOperation(row[0]);
+        FruitTransaction fruitTransaction = new FruitTransaction();
+        fruitTransaction.setOperation(operation);
+        fruitTransaction.setFruit(row[1]);
+        fruitTransaction.setQuantity(Integer.parseInt(row[2]));
+        return fruitTransaction;
+    }
+
+    private void writeLinesToFile(String lines, String pathToFile) {
+        Optional.ofNullable(lines)
+                .orElseThrow(() -> new IllegalArgumentException(STRING_ARGUMENT_ERROR));
+        try (OutputStream outputStream = new FileOutputStream(pathToFile);
+                OutputStreamWriter outputStreamWriter = new OutputStreamWriter(outputStream);
+                BufferedWriter writer = new BufferedWriter(outputStreamWriter)) {
```

Comment:
```suggestion
        try (BufferedWriter writer = new BufferedWriter(new FileReader(pathToFile))) {
```

Reply:
BufferedWriter must receive an object (heirs) of the Writer class in the constructor argument, but in this case the argument is an instance of the FileReader class. so it doesn't work. Or I don't know how to refactor it.